### PR TITLE
ENT-3723: Rebranding update: more robust get_platform_logo_url and Learner Portal default colors update

### PR DESCRIPTION
**Pull request by chavesj**
**ENT-3723:**

As part of the rebranding effort, `get_logo_url` is being updated. Since `get_logo_url` will always return an absolute URL, we can remove the use of `urljoin` in `get_platform_logo_url`

[Associated edx-platform PR](https://github.com/edx/edx-platform/pull/25654)

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [x] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [x] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/edx-enterprise), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise/utils.py [hunk 1]
```diff
@@ -1446,10 +1445,7 @@ def get_platform_logo_url():
     Return an absolute URL of the platform logo using the branding api
     """
     # Return fake URL for tests rather than mock get_logo_url for every test using EnterpriseCustomer
-    return urljoin(
```
Comment (awaisdar001) [inline enterprise/utils.py:1449]: I would just want to have what was the reason it was added in the first place? Just wanted to make sure we are not missing an edge case.

**Review by iloveagent57 — COMMENTED** [review]
Let's have a unit test for this?  At least in a follow-on PR, if you're trying to ship this quickly.

File: enterprise/utils.py (hunk 1 above)
Comment (chavesj) [reply enterprise/utils.py:1449]: I believe this was done in this way because `get_logo_url` returned a relative path to the platform logo URL at time of implementation and the domain of that URL was the LMS base URL.

Comment (awaisdar001) [reply enterprise/utils.py:1449]: just wanted to throw this question, can you tell if openedX installations are using edx-enterprise? i think the answer is no, but just wanted to ask.

**Review by iloveagent57 — APPROVED** [review]

**Review by adamstankiewicz — APPROVED** [review]

File: enterprise/utils.py (hunk 1 above)
Comment (chavesj) [reply enterprise/utils.py:1449]: Here's the [devstack setting](https://github.com/edx/edx-platform/blob/master/lms/envs/devstack.py#L350) to enable/disable enterprise integration (use of edx-enterprise). Enterprise integration is currently enabled by default in devstack.

File: enterprise/utils.py [hunk 2]
```diff
@@ -1446,10 +1445,7 @@ def get_platform_logo_url():
...
-        settings.LMS_ROOT_URL,
-        get_logo_url()
-    ) if get_logo_url else 'http://fake.url'
+    return get_logo_url() if get_logo_url else 'http://fake.url'
```
Comment (awaisdar001) [inline enterprise/utils.py:1448]: Can we make this method a little robust using `urlparse`?
here is what i mean. 

```
from urllib.parse import urlparse


def is_valid_url(url):
  try:
    result = urlparse(url)
    return all([result.scheme, result.netloc])
  except ValueError:
    return False

fake_url = 'http://fake.url'
logo_url = get_logo_url() 
if not logo_url:
    return fake_url

is_abs_url = is_valid_url(logo_url)
if is_abs_url:
    return logo_url

return urljoin(settings.LMS_ROOT_URL, logo_url)
```

the reason I say that there is a good chance that a logo can be overridden using site_configuration with either `logo_image_url` key or `university` key. so in that case, we can still get a relative URL (which would need urljoin).

Comment (awaisdar001) [reply enterprise/utils.py:1448]: if we do this way, then this PR shouldn't be blocked by the edx-platform PR.

File: enterprise/utils.py [hunk 3]
```diff
@@ -1451,15 +1450,29 @@ def enroll_users_in_course(
...
     """
     # Return fake URL for tests rather than mock get_logo_url for every test using EnterpriseCustomer
-    return urljoin(
-        settings.LMS_ROOT_URL,
-        get_logo_url()
-    ) if get_logo_url else 'http://fake.url'
+    fake_url = 'http://fake.url'
+    logo_url = get_logo_url()
+    if not logo_url:
+        return None
```
Comment (awaisdar001) [inline enterprise/utils.py:1469]: would like to return `fake_url` url here?

Comment (adamstankiewicz) [reply enterprise/utils.py:1469]: Based on the comment "Return fake URL for tests rather than mock get_logo_url for every test using EnterpriseCustomer", I think the `fake_url` was only included here for tests? I think it would be odd to actually return `http://fake.url` for real as that's not a valid URL, so might as well just return `None`.

I will remove that comment now that the test has been updated.

Comment (awaisdar001) [reply enterprise/utils.py:1467]: Ok got it.
